### PR TITLE
network

### PR DESCRIPTION
**Pull request by ownluke**
Implementation network interface: p2pv1 & p2pv2

### CODE REVIEW

File: kernel/network/context/context.go [hunk 1]
```diff
@@ -15,30 +16,108 @@ import (
...
+// 操作级上下文，不是所有的方法都需要独立上下文，按需选用
+type OperateCtx interface {
+	context.Context
+
+	GetLog() logs.Logger
+	GetTimer() *timer.XTimer
+	IsValid() bool
+}
+
+type OperateCtxImpl struct {
```
Comment (chunhui01) [inline kernel/network/context/context.go:89]: There is no special requirement to use xcontext.BaseCtx directly.

File: kernel/network/config/config.go [hunk 2]
```diff
@@ -1,125 +1,135 @@
...
+    "fmt"
+    "github.com/spf13/viper"
+    utils "github.com/xuperchain/xupercore/kernel/common/xutils"
+)
 
-	"github.com/spf13/viper"
+// default settings
+const (
+    DefaultNetPort          = 47101             // p2p port
+    DefaultNetKeyPath       = "./data/netkeys" 	// node private key path
```
Comment (chunhui01) [inline kernel/network/config/config.go:12]: Absolute path is recommended for file path,It can be obtained through envconf.

**Review by chunhui01 — APPROVED** [review]
LGTM

File: bcs/network/p2pv1/client.go [hunk 3]
```diff
@@ -0,0 +1,183 @@
+package p2pv1
+
+import (
+    "errors"
```
Comment (preminem) [inline bcs/network/p2pv1/client.go:4]: When importing multiple packages, the packages should be grouped. There is no need for a blank line between packages in the same group, and a blank line between packages in different groups. 
Standard library packages should be placed in the first group, third-party library should be placed in the second group, and local library should be placed in the third group.

**Review by preminem — CHANGES_REQUESTED** [review]

File: kernel/network/network.go [hunk 4]
```diff
@@ -58,33 +58,26 @@ func createP2PServ(name string) nbase.P2PServer {
...
 // 暂时没有特殊的逻辑，先简单透传，预留方便后续扩展
 type NetworkImpl struct {
-	netCtx  *nctx.NetCtx
-	p2pServ nbase.P2PServer
+	ctx    nctx.DomainCtx
+	p2pServ p2p.Server
 }
 
-func CreateNetwork(servName string, netCtx *nctx.NetCtx) (Network, error) {
+func CreateNetwork(ctx nctx.DomainCtx, servName string) (Network, error) {
```
Comment (preminem) [inline kernel/network/network.go:78]: servName may be redundant, that included in the p2p conf of the network context

**Pull request merged**